tests/scale/virt-api.go

```diff
@@ -25,29 +29,21 @@ var _ = Describe("[sig-compute] virt-api scaling", decorators.SigCompute, func()
 	numberOfNodes := 0
 
 	setccs := func(ccs v12.CustomizeComponents) (oldcss v12.CustomizeComponents) {
-		originalKv := util.GetCurrentKv(virtClient)
-		kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(context.Background(), originalKv.Name, v1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
+		kv := util.GetCurrentKv(virtClient)
 		oldcss = kv.Spec.CustomizeComponents
-		kv.Spec.CustomizeComponents = ccs
-		EventuallyWithOffset(1, func() error {
-			_, err = virtClient.KubeVirt(originalKv.Namespace).Update(context.Background(), kv, v1.UpdateOptions{})
-			return err
-		}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
-
+		patchPayload, err := patch.New(patch.WithReplace("/spec/customizeComponents", ccs)).GeneratePayload()
+		Expect(err).ToNot(HaveOccurred())
+		_, err = virtClient.KubeVirt(kv.Namespace).Patch(context.Background(), kv.GetName(), types.JSONPatchType, patchPayload, v1.PatchOptions{})
+		Expect(err).ToNot(HaveOccurred())
 		return oldcss
 	}
 
 	restorescc := func(ccs v12.CustomizeComponents) {
-		originalKv := util.GetCurrentKv(virtClient)
-		kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(context.Background(), originalKv.Name, v1.GetOptions{})
+		kv := util.GetCurrentKv(virtClient)
```

Comment:
I don't think that `kv` is actually used by this function.

Reply:
Not sure what did you mean, the kv variable is used to get the namespace (kv.Namespace) and the name (kv.GetName()) needed for the patch operation. Both are essential for identifying the resource to patch.

Reply:
You are right. However, the namespace and name of the resource are constant so it does not have to be read both in getscc and in restore.